src/Configuration/src/ConfigServerBase/ConfigServerConfigurationProvider.cs

```diff
@@ -925,13 +925,13 @@ protected internal string FetchAccessToken()
             return null;
         }
 
-        return HttpClientHelper.GetAccessToken(settings.AccessTokenUri, settings.ClientId, settings.ClientSecret, settings.Timeout,
+        return HttpClientHelper.GetAccessTokenAsync(settings.AccessTokenUri, settings.ClientId, settings.ClientSecret, settings.Timeout,
             settings.ValidateCertificates, httpClient, logger).GetAwaiter().GetResult();
     }
 
     // fire and forget
 #pragma warning disable S3168 // "async" methods should not return "void"
-    protected internal async void RefreshVaultTokenAsync(object state)
+    protected internal async void RefreshVaultToken(object state)
```

Comment:
While it doesn't return `Task`, shouldn't it retain `async` anyway given that it is async?

Reply:
* This is not the only instance, but I'll refrain from commenting on any others until we resolve here

Reply:
I've removed the Async suffix in all three suggested cases, for the next reason. The purpose of the `async` keyword in the method declaration is only for the method body itself. It is unobservable by callers, they see just `void` or `Task`. So from a callers perspective, this method has nothing to do with asynchronous execution, which is why adding the prefix would be wrong.